test/filtering/filter_test.py

```diff
@@ -1,30 +1,98 @@
-    
-    def test_filter_input_streams_equals(self):
-        self.assertEqual(["A","B"], self.f.get_inputs())
+        self.assertEqual(2, len(self.f.get_input_names()))
+
+    def test_filter_input_stream_names_equals(self):
+        self.assertEqual(["A", "B"], self.f.get_input_names())
 
     def test_filter_output_number_correct(self):
-        self.assertEqual(3, len(self.f.get_outputs()))
+        self.assertEqual(3, len(self.f.get_output_names()))
+
+    def test_filter_output_stream_names_equals(self):
+        self.assertEqual(["D", "E", "F"], self.f.get_output_names())
+
+    def test_get_in_streams(self):
+        self.assertEqual(self.s_A, self.f._get_input(0))
+        self.assertEqual(self.s_B, self.f._get_input(1))
+
+    def test_get_out_streams(self):
+        self.assertEqual(self.s_D, self.f._get_output(0))
+        self.assertEqual(self.s_E, self.f._get_output(1))
+        self.assertEqual(self.s_F, self.f._get_output(2))
+
+    def test_get_in_iters(self):
+        self.assertEqual(iter(self.s_A), self.f._get_in_iter(0))
+        self.assertEqual(iter(self.s_B), self.f._get_in_iter(1))
+
+    def test_get_out_iters(self):
+        self.assertEqual(iter(self.s_D), self.f._get_out_iter(0))
+        self.assertEqual(iter(self.s_E), self.f._get_out_iter(1))
+        self.assertEqual(iter(self.s_F), self.f._get_out_iter(2))
+
+    def test_pop_data(self):
+        self.assertEqual(1, self.f._pop_data(0))
+        self.assertEqual(3, self.f._pop_data(1))
+
+    def test_push_data(self):
+        self.f._push_data(5, 0)
+        self.assertEqual(5, self.s_D.__iter__().__next__())
+
+    def test_execute_outputs_closed(self):
+        self.s_D.close()
+        self.s_E.close()
+        self.s_F.close()
+        self.f._on_outputs_closed = MagicMock()
+        self.f.execute()
+        assert self.f._on_outputs_closed.called
+
+    def test_execute_on_data(self):
+        self.f._on_data = MagicMock()
+        self.f.execute()
+        assert self.f._on_data.called
+
+    def test_execute_input_empty(self):
+        self.s_A.clear()
+        self.s_B.clear()
+        self.f._on_inputs_empty = MagicMock()
+        self.f.execute()
+        assert self.f._on_inputs_empty.called
 
-    def test_filter_execute_raise_exception(self):
-        self.assertRaises(NotImplementedError, self.f.execute)
+    def test_execute_input_closed(self):
+        self.s_A.clear()
+        self.s_B.clear()
+        self.s_A.close()
+        self.s_B.close()
+        self.f._on_inputs_closed = MagicMock()
+        self.f.execute()
+        assert self.f._on_inputs_closed.called
 
-    def test_filter_setup_raise_exception(self):
-        self.assertRaises(NotImplementedError, self.f.setup, [],[], [])
\ No newline at end of file
+    def test_default_on_inputs_closed_closes_outputs(self):
+        self.s_A.clear()
+        self.s_B.clear()
+        self.s_A.close()
+        self.s_B.close()
+        self.f.execute()
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5775588)